src/mofaflex/_core/mofaflex.py

```diff
@@ -785,9 +904,42 @@ def _fit(self, data, preprocessor):
                 for scale in prior_scales.values():
                     scale[: self._n_dense_factors, :] = dense_scale
 
+        # guided factors
+        guiding_vars = GuidingVarsDataset(
+            data, self._data_opts.guiding_vars_obs_keys, self._model_opts.guiding_vars_likelihoods
+        )
         covariates = CovariatesDataset(data, self._data_opts.covariates_obs_key, self._data_opts.covariates_obsm_key)
+
+        init_tensor = self._initialize_factors(data)
+
+        # add non-initialized factors for guiding variables to init_tensor
+        guiding_vars_names = (
+            self._data_opts.guiding_vars_obs_keys.keys() if self._data_opts.guiding_vars_obs_keys else []
+        )
+        if len(guiding_vars_names) > 0:
+            for group_name in self._group_names:
+                for key, value in [("loc", 0), ("scale", self._model_opts.init_scale)]:
+                    init_tensor[group_name][key] = torch.cat(
+                        [
+                            value
+                            * torch.ones(
+                                len(guiding_vars_names),
+                                1,
+                                init_tensor[group_name][key].shape[-1],
+                                device=self._train_opts.device,
+                            ),
+                            init_tensor[group_name][key],
+                        ],
+                        axis=0,
+                    )
```

Comment:
I feel like this belongs into `_initialize_factors`. That method already handles informed factors.